test/common/testvm.py

```diff
@@ -259,7 +273,8 @@ def _kill_ssh_master(self):
             self.ssh_master = None
         if self.ssh_process:
             self.ssh_process.stdin.close()
-            self.ssh_process.wait()
+            if not self._abort:
+                self.ssh_process.wait()
```

Comment:
What I don't understand is why the behavior with a timeout is different from the behavior with any other failure. Why do we have to avoid killing the ssh master in that case?